library/src/cache/patch_manager.rs

```diff
@@ -52,11 +55,19 @@ pub trait ManagePatches {
     /// or None if no patch is installed.
     fn last_successfully_booted_patch(&self) -> Option<PatchInfo>;
 
+    /// The patch we most recently attempted to boot. Will be the same as
+    /// last_successfully_booted_patch if the last boot was successful.
+    fn last_attempted_boot_patch(&self) -> Option<PatchInfo>;
```

Comment:
Looks like PatchInfo and PatchMetadata are the same thing, just with internal/external types? https://github.com/shorebirdtech/updater/blob/0a0cd183f529cb9f2bab4d8ef63c893dd9f3a684/library/src/cache/mod.rs#L9

Reply:
Not quite. The naming could certainly be better, but PatchInfo has a number and a path, PatchMetadata has a number and a size. PatchMetadata is internal to PatchManager, PatchInfo is the interface used by other modules to talk to PatchManager. We could probably consolidate these, but that would probably be better as a future refactor.